### PR TITLE
[TIMOB-24071] iOS: Expose Ti.UI.Label maxLines (Parity)

### PR DESCRIPTION
**Pull request by chmiiller**
**JIRA:** https://jira.appcelerator.org/browse/TIMOB-24071

### CODE REVIEW

**Comment by sgtcoolguy** [comment]
ok to test

File: apidoc/Titanium/UI/Label.yml [hunk 1]
```diff
@@ -157,10 +157,10 @@ properties:
     since: "5.4.0"
         
   - name: maxLines
-    summary: Makes the label at most this many lines tall.
-    platforms: [android]
+    summary: Makes the label at most this many lines tall. 
+    platforms: [android, iphone, ipad]
     type: Number
-    since: "4.1.0"
+    since: {android: "4.1.0", ios: "6.0.0"}
```
Comment (hansemannn) [inline apidoc/Titanium/UI/Label.yml]: Rename to  `{android: "4.1.0", iphone: "6.1.0", ipad: "6.1.0}`
- 6.0.0 is already and Beta and will not take new features at this point
- The docs need to distinguish `iphone` and `ipad` as specified keys

Comment (hansemannn) [inline apidoc/Titanium/UI/Label.yml]: Rename to  `{android: "4.1.0", iphone: "6.1.0", ipad: "6.1.0}`
- 6.0.0 is already and Beta and will not take new features at this point
- The docs need to distinguish `iphone` and `ipad` as specified keys

File: iphone/Classes/TiUILabel.m [hunk 2]
```diff
@@ -420,6 +420,14 @@ -(void)setVerticalAlign_:(id)value
         [self padLabel];
     }
 }
+
+-(void)setMaxLines_:(id)value
+{
+    [[self label] setNumberOfLines:[TiUtils floatValue:value]];
+    [self padLabel];
+	[(TiViewProxy *)[self proxy] contentsWillChange];
```
Comment (hansemannn) [inline iphone/Classes/TiUILabel.m]: The indentation is messed-up here, please align with the other lines (use diff-tool to check before commiting).

File: iphone/Classes/TiUILabel.m [hunk 3]
```diff
@@ -420,6 +420,14 @@ -(void)setVerticalAlign_:(id)value
         [self padLabel];
     }
 }
+
+-(void)setMaxLines_:(id)value
+{
+    [[self label] setNumberOfLines:[TiUtils floatValue:value]];
```
Comment (hansemannn) [inline iphone/Classes/TiUILabel.m]: Use `ENSURE_TYPE(value, NSNumber);` for validation

**Review by hansemannn — CHANGES_REQUESTED** [review]
Left some comments that require changes. Please also add unit-test for this property in https://github.com/appcelerator/titanium-mobile-mocha-suite

**Comment by sgtcoolguy** [comment]
@chmiiller regarding Hans' comment about our unit tests. We maintain a test suite now that we use against PRs. The suite itself is at https://github.com/appcelerator/titanium-mobile-mocha-suite

In particular, it'd be good to add a unit test for Ti.UI.Label.setMaxLines in https://github.com/appcelerator/titanium-mobile-mocha-suite/blob/master/Resources/ti.ui.label.test.js

To do so, clone that repo down locally. Copy titanium-mobile-mocha-suite/Resources/ti.ui.label.test.js to titanium_mobile/tests/Resources/ti.ui.label.test.js and then edit it to add a test for maxLines. (Anything in the titanium_mobile/tests folder acts as an "override" for the test suite. So you could also cheat and copy the app.js and comment out all the other suites other than ti.ui.label.test.)

You can run the test suite with that change locally by doing:
```
cd titanium_mobile/build
npm install
node scons.js cleanbuild ios # builds and installs ios-only SDK as 6.1.0 in your system
node scons.js test iOS # runs the unit tests against iOS
```

**Comment by hansemannn** [comment]
@chmiiller Thanks for the changes! Unfortunately you made the changes in this repo instead of [titanium-mobile-mocha-suite](https://github.com/appcelerator/titanium-mobile-mocha-suite). Can you just push https://github.com/appcelerator/titanium_mobile/pull/8543/commits/b2b706cd12085a32938f5b780ea19f65d23b5e1e to the existing repo and I will review the PR there as well. Thx!

**Comment by chmiiller** [comment]
ops, I got bit confused by @sgtcoolguy 's answer. Gonna push it...

**Comment by chmiiller** [comment]
here it is: https://github.com/appcelerator/titanium-mobile-mocha-suite/pull/6
Hope it is correct now 😁

**Review by hansemannn — APPROVED** [review]

**Comment by hansemannn** [comment]
CR + FT **passed**, PR **approved**. Thanks @chmiiller! 🚀

**Comment by chmiiller** [comment]
this [commit](https://github.com/appcelerator/titanium_mobile/pull/8543/commits/74131e3dbd795f081f935f064fe8dc96b748e8e8) removes the ti.ui.label.test.js added by mistake in the repo

**Comment by sgtcoolguy** [comment]
cc @hansemannn and @chmiiller : Just to clear up the confusion here - We typically add the new/modified unit tests in our PRs as you did, @chmiiller - that way we can ensure that the test is run and passes as part of the PR that modifies the behavior. Then once merged, we migrate the changes back to the titanium-mobile-mocha-suite so that the test is in the canonical suite used by all platforms. It's a fairly new process here and not that friendly yet, which is why even we are a little iffy about it. 😏 We'd eventually like to have some auto-merge process that once a test gets modified as part of a PR in titanium_mobile and is merged in and passes it'd get auto-merged back into the suite repo.

As for why we place the modified test in the PR itself? Otherwise once it goes into the full suite it'll get run against every PR, so totally unrelated PRs would suddenly start failing because of the modified test without the PR that implements/fixes it being merged yet.

**Comment by chmiiller** [comment]
Makes (a lot of) sense now =)
Thanks!

**Comment by yomybaby** [comment]
👍

**Pull request merged**